test/contract/deploy_contract_test.exs

```diff
@@ -138,6 +138,19 @@ defmodule Soroban.Contract.DeployContractTest do
      }} = DeployContract.deploy(wasm_id, source_secret)
   end
 
+  test "retrieve_xdr_to_sign/2", %{
+    wasm_id: wasm_id,
+    source_public_with_error: source_public
+  } do
+    assert String.contains?(
+             DeployContract.retrieve_unsigned_xdr_to_deploy(
+               wasm_id,
+               source_public
+             ),
+             "AAAAAgAAAAAlj"
+           )
```

Comment:
To be more precise with tests we could avoid the `String.contains?`.

My 5 cents:
- Create a variable `tx_envelope_xdr` in the setup section.
- Do pattern matching in the test

Reply:
@miguelnietoa In this case we can't do it like that, because when we use 
``` 
    function =
      HostFunction.new(
        type: :create,
        wasm_id: wasm_id
      )
```
it uses a random salt that will change the xdr envelope every time the function is executed

Reply:
WDYT about mocking the :crypto library? 🤔